libraries/kubelet.rb

```diff
@@ -5,55 +5,81 @@ module KubernetesCookbook
   class KubeletService < Chef::Resource
     resource_name :kubelet_service
 
+    property :version, String, default: '1.7.6'
     property :remote, String,
-      default: 'https://storage.googleapis.com/kubernetes-release' \
-               '/release/v1.4.0/bin/linux/amd64/kubelet'
+      default: lazy { |r|
+        'https://storage.googleapis.com/kubernetes-release' \
+        "/release/v#{r.version}/bin/linux/amd64/kubelet"
+      }
     property :checksum, String,
-      default: '531dad0b316d214923d3e928f6ebc3c6' \
-               'be3a7f37c3f5a453e02807c22d7c0fd4'
+      default: '6178cb17d3c34ebe31dfc572d17ae077ce19d2a936bbe90999bac87ebf6e06eb'
     property :container_runtime_service, String, default: 'docker.service'
     property :run_user, String, default: 'kubernetes'
+    property :file_ulimit, Integer, default: 65536
 
     # Reference: http://kubernetes.io/docs/admin/kubelet/
-    property :api_servers, default: nil
+    property :api_servers
 
     default_action :create
 
     action :create do
-      remote_file 'kubelet binary' do
+      remote_file "kubelet binary version: #{new_resource.version}" do
         path kubelet_path
         mode '0755'
         source new_resource.remote
         checksum new_resource.checksum
       end
+
+      pkgs = case node['platform_family']
+             when 'debian'
+               %w(iptables iproute2 socat util-linux mount ebtables)
```

Comment:
I think `ethtool` is still needed since https://github.com/kubernetes/kubernetes/issues/26093 is still open and https://github.com/kubernetes/kubernetes/pull/30356 was never merged.